modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationManager.java

```diff
@@ -241,4 +250,91 @@ private void validate(ServiceTemplateEntity serviceTemplateEntity,
                 configurationParameters, serviceConfigurationUpdate.getConfiguration(),
                 jsonObjectSchema);
     }
+
+    /**
+     * Query pending configuration change request for agent.
+     */
+    public ResponseEntity<ConfigurationChangeRequestForAgent>
+            getPendingConfigurationChangeRequest(String serviceId, String resourceName) {
+        ServiceConfigurationUpdateRequestQueryModel model =
+                new ServiceConfigurationUpdateRequestQueryModel(null, UUID.fromString(serviceId),
+                        resourceName, null, ServiceConfigurationStatus.PENDING);
+        List<ServiceConfigurationUpdateRequest> requests =
+                serviceConfigurationUpdateStorage.listServiceConfigurationUpdateRequests(model);
```

Comment:
It better to order the list by `startedTime` desc in the query method `listServiceConfigurationUpdateRequests` and then getFrist() if the list not empty to get the latest request. This is more efficient than using stream().max().